### ISSUE TITLE
Lint/MissingCopEnableDirective should recognise # rubocop:push

### ISSUE THREAD

**Issue by timbaverstock-bmbl** [rubocop/rubocop#15136]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/MissingCopEnableDirective

### Steps to reproduce the problem

1. Create a file:
```
# frozen_string_literal: true

def test
  # rubocop:push
  # rubocop:disable Style/RescueStandardError
rescue => e
  print e
end
# rubocop:pop
```
2. Run rubocop

### Expected behavior

No warning while within a push/pop pair: the pop reinstates the status quo ante.

### Actual behavior

Inspecting 1 file
W

Offenses:

test.rb:5:3: W: Lint/MissingCopEnableDirective: Re-enable Style/RescueStandardError cop with # rubocop:enable after disabling it.
  # rubocop:disable Style/RescueStandardError
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected

### RuboCop configuration

```yaml
AllCops:
  NewCops: enable
```

### RuboCop version

```shell
% bundle exec rubocop -V     
1.86.1 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.3.6) [arm64-darwin24]
```
